docs/enhancements/fep-001-api-fleet-templating.md

```diff
@@ -0,0 +1,300 @@
+As an IT administrator, I want the name of my MicroShift instance in ACM to have the same cluster name as the underlying device's name or fingerprint, so I can easily correlate between the two.
+
+#### Story 5: site specific telemetry endpoints
+As an IT administrator, I want to provide site-specific/region-specific telemetry endpoints to devices.
+
+#### Story 6: dialing up verbosity for logging in my user application in specific devices
+As an IT administrator, I want to dial up the verbosity for logging on a specific device, so I can troubleshoot it.
+
+#### Story 7: region specific application service endpoints
+As an IT administrator, I want to provide region-specific service endpoints to my applications
+
+#### Story 8: region specific radio bands/regulations
+As a developer, I want to ensure that my devices automatically comply with local regulations when enrolled in specific regions. (i.e. configured with region-specific radio bands for WiFi, 5G, or other radio technologies)
+
+#### Story 9: region specific data sovereignty
+As a developer, I want to ensure that my devices connect to the specific regional endpoints to ensure compliance with data sovereignty.
+
+### Notes/Constraints/Caveats
+
+We would need to identify the design changes relative to templateVersions and
+if the relationship of those continues to be to the fleet or device level.
+
+Traceability of device spec changes over time becomes increasingly important with templating,
+as devices will render the TemplateVersions in different ways over time, we probably
+need an audit log where we can record how devices are being rendered over time.
+
+Would it suffice to have the ability to send those specific traces to a logging system?
+Keeping those in the database would probably be dangerous over time, specially at the scales
+we intend this system to work.
+
+
+### Risks and Mitigations
+
+When rolling out a configuration to a fleet of devices, during the rollout, some devices could turn out to be
+not renderable, for example:
+* due to a missing label
+* because the template rendering is referencing non existing resources
+
+This is partially mitigated by the initial device scan performed by the fleet controller
+when a fleet.spec.template changes. Still, labels could change while a fleet is being rolled out.
+
+We probably need to differentiate between "fleet controller did not find the label, so
+failed to do the substitution" and "the referenced resource (after substitution) does
+not exist / isn't accessible". There is little the service can do about the latter:
+verifying service-side is costly and also futile, because the resource could be there when
+the service tried, but no longer there when the agent needs it. At the end of the day,
+the agent will try during staging, fail, and set its condition to "failed to reconcile".
+The former case would be similar (failure to reconcile). Then, if errors exceed threshold,
+stop the rollout.
```

Comment:
Removing this paragraph in favor of a more specific implementation in sections: 
#### Expected behavior on issues
##### Device cannot be rendered from Template due to missing label
##### Device cannot generate /device/{name}/rendered version due to missing resource